### PR TITLE
[Impeller] re-enable AHB swapchain.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/148139

Block on the CPU rather than immediately beginning encoding. Seems to fix the synchronization issues that we knew about. We need to find a way to use AHB swapchains, as it seems like correct synchronization via either SurfaceView or a SurfaceControl hierarchy will require exposing our rendering as a SurfaceTransaction.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This is primarily tested via eyeballing and looking at performance benchmarks. We can't really unit test the timing/fence synchronization.

File: impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc [hunk 1]
```diff
@@ -15,14 +15,15 @@
 #include "impeller/renderer/backend/vulkan/swapchain/surface_vk.h"
 #include "impeller/toolkit/android/surface_transaction.h"
 #include "impeller/toolkit/android/surface_transaction_stats.h"
+#include "vulkan/vulkan_enums.hpp"
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc:18]: Bad include? `impeller/renderer/backend/vulkan/vk.h` should be the right include.

Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc:18]: Fixed

Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc]: ...now

**Review by chinmaygarde — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Unfortunately I found that this does not currently work with hybrid composition, we fail to construct the SurfaceControl

**Comment by chinmaygarde** [comment]
The only failure mode I am aware of is ASurfaceControl_createFromWindow. Do we not have a window? If not, we can create a control and add it to an existing control reference.

**Comment by jonahwilliams** [comment]
We have the Surface/window obtained from the Image reader but ASurfaceControl_createFromWindow is returning nullptr. 

I'm looking into reparenting for hc++ but that will require API 34 to get the correct behavior for the platform views themselves

**Comment by jonahwilliams** [comment]
The failure case here is that the SurfaceControl object is invalid (because the parenting failed). This is easy to detect and we can fall back to the khr swapchain.

**Pull request merged**